### ISSUE TITLE
Add ZIO.fromIterableEffect + fix inconsistent method naming

### ISSUE THREAD

**Issue by mindartur** [zio/zio#3811]
Since ZStream is basically a stream of effects, why not add the following method? 
Correct me if it is already implemented somewhere:

```
def fromIterableEffect[R, E, A](as: => Iterable[ZIO[R, E, A]]): ZStream[R, E, A] 
```

Also naming is a bit inconsistent in a following existing methods:
```
def fromIterableM[R, E, O](iterable: ZIO[R, E, Iterable[O]]): ZStream[R, E, O]
```
and
```
def fromIteratorEffect[R, A](
    iterator: ZIO[R, Throwable, Iterator[A]]
  ): ZStream[R, Throwable, A]
```

**Comment by mindartur** [comment]
I'll be happy to fix those, if it makes sense :)

**Comment by iravid** [comment]
@mindartur I tend to think that `Iterable[ZIO]` is a rare enough case to not justify a dedicated constructor, because you can fairly easily do `ZStream.fromIterable(as).mapM(identity)`. What do you think?

`fromIterableM` for is indeed a shoddy name. How about `fromEffectIterable` for that one?

**Comment by mindartur** [comment]
> I tend to think that Iterable[ZIO] is a rare enough case to not justify a dedicated constructor, because you can fairly easily do ZStream.fromIterable(as).mapM(identity). What do you think?

Agree, if it is not used often, then no need to implement it. I actually solved my problem the same way: `ZStream.fromIterable(as).mapM(identity)`.

> fromIterableM for is indeed a shoddy name. How about fromEffectIterable for that one?

tricky here, `fromEffectIterable` looks better, but `fromIterableM` reminds of `mapM` and`map` methods pair :)

**Comment by iravid** [comment]
Pinging @regiskuckaertz @simpadjo @adamgfraser for thoughts on renaming `ZStream.fromIterableM` to `fromEffectIterable`.

I admit I was looking for something like `fromIterableM` the other day but couldn't find it because I was looking under `fromEffect`.

**Comment by simpadjo** [comment]
I don't have an opinion on that
